examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -315,13 +315,20 @@ on_event(struct pt_regs* ctx) {
 
     // Are we in user mode?
     if (cs & 3) {
+      // Yes - use the registers context given to the BPF program
       user_regs = *ctx;
     }
     else {
+      // No - use the registers context of usermode, that is stored on the stack.
+
       // The third argument is equivalent to `task_pt_regs(task)` for x86. Macros doesn't
       // work properly on bcc, so we need to re-implement.
       bpf_probe_read_kernel(
           &user_regs, sizeof(user_regs),
+          // Note - BCC emits an implicit bpf_probe_read_kernel() here (for the deref of 'task').
+          // I don't like the implicitness (and it will be something we'll need to fix if we're ever
+          // to move from BCC). But meanwhile, I tried to change it to explicit and the BPF assembly
+          // varies to much as I prefer to avoid this change now ;(
```

Comment:
Something feels wrong with this sentence. Maybe
```suggestion
          // to move from BCC). Meanwhile, I tried to change it to explicit but the BPF assembly
          // varies to much so I prefer to avoid this change now ;(
```
?

Reply:
Yeah agree that sentence is better